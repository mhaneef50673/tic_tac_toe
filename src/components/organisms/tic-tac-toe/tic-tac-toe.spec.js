import React from "react";
import { mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";
import TicTacToe from "./index";

describe("TicTacToe Component", () => {
  it("should render the component", () => {
    const tree = mount(<TicTacToe />);
    expect(mountToJson(tree)).toMatchSnapshot();
  });
  it("should change setState when onGameOver change is called", () => {
    const tree = mount(<TicTacToe />);
    const instance = tree.instance();
    expect(instance.state.leaderBoard).toEqual([]);
    instance.onGameOver("player1", "player2", "player1");
    instance.forceUpdate();
    expect(instance.state.leaderBoard).toEqual([
      { name: "player1", points: 1 },
      { name: "player2", points: 0 },
    ]);
  });
});
