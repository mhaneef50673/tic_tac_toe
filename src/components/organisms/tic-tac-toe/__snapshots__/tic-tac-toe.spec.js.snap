// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicTacToe Component should render the component 1`] = `
<TicTacToe>
  <Container
    className="main-container"
    fluid={true}
  >
    <div
      className="main-container container-fluid"
    >
      <h1>
        Tic Tac Toe
      </h1>
      <Row
        noGutters={false}
      >
        <div
          className="row"
        >
          <Col
            lg={6}
          >
            <div
              className="col-lg-6"
            >
              <GameWidget
                onGameOver={[Function]}
              >
                <div
                  className="game-widget"
                >
                  <div
                    className="panel-wrapper"
                  >
                    <div
                      className="panel-container"
                    >
                      <h2>
                        Game
                      </h2>
                      <div
                        className="game-widget_container"
                      >
                        <Row
                          noGutters={false}
                        >
                          <div
                            className="row"
                          >
                            <Col>
                              <div
                                className="col"
                              >
                                <GameForm
                                  isError={false}
                                  onGameStartHandler={[Function]}
                                  onInputChange={[Function]}
                                  playerOne=""
                                  playerTwo=""
                                  shouldStartGame={false}
                                >
                                  <form
                                    onSubmit={[Function]}
                                  >
                                    <label
                                      className="input-label"
                                    >
                                      Player one Name
                                    </label>
                                    <input
                                      className="input-text"
                                      onChange={[Function]}
                                      placeholder="Enter player one name"
                                      required={true}
                                      type="text"
                                      value=""
                                    />
                                    <label
                                      className="input-label"
                                    >
                                      Player two Name
                                    </label>
                                    <input
                                      className="input-text"
                                      onChange={[Function]}
                                      placeholder="Enter player two name"
                                      required={true}
                                      type="text"
                                      value=""
                                    />
                                    <div
                                      className="btn-center "
                                    >
                                      <button
                                        className="btn-class"
                                        type="submit"
                                      >
                                        Start Game
                                      </button>
                                    </div>
                                  </form>
                                </GameForm>
                              </div>
                            </Col>
                          </div>
                        </Row>
                      </div>
                    </div>
                  </div>
                </div>
              </GameWidget>
            </div>
          </Col>
          <Col
            lg={6}
          >
            <div
              className="col-lg-6"
            >
              <GameResult
                isError={false}
                isFetching={true}
                results={Array []}
              >
                <div
                  className="game-results"
                >
                  <div
                    className="panel-wrapper"
                  >
                    <div
                      className="panel-container"
                    >
                      <h2>
                        Leaderboard
                        <span
                          className="trophy-icon"
                        >
                          <FontAwesomeIcon
                            border={false}
                            className=""
                            color="#FFD700"
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  576,
                                  512,
                                  Array [],
                                  "f091",
                                  "M552 64H448V24c0-13.3-10.7-24-24-24H152c-13.3 0-24 10.7-24 24v40H24C10.7 64 0 74.7 0 88v56c0 35.7 22.5 72.4 61.9 100.7 31.5 22.7 69.8 37.1 110 41.7C203.3 338.5 240 360 240 360v72h-48c-35.3 0-64 20.7-64 56v12c0 6.6 5.4 12 12 12h296c6.6 0 12-5.4 12-12v-12c0-35.3-28.7-56-64-56h-48v-72s36.7-21.5 68.1-73.6c40.3-4.6 78.6-19 110-41.7 39.3-28.3 61.9-65 61.9-100.7V88c0-13.3-10.7-24-24-24zM99.3 192.8C74.9 175.2 64 155.6 64 144v-16h64.2c1 32.6 5.8 61.2 12.8 86.2-15.1-5.2-29.2-12.4-41.7-21.4zM512 144c0 16.1-17.7 36.1-35.3 48.8-12.5 9-26.7 16.2-41.8 21.4 7-25 11.8-53.6 12.8-86.2H512v16z",
                                ],
                                "iconName": "trophy",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            swapOpacity={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-trophy fa-w-18 "
                              color="#FFD700"
                              data-icon="trophy"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 576 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M552 64H448V24c0-13.3-10.7-24-24-24H152c-13.3 0-24 10.7-24 24v40H24C10.7 64 0 74.7 0 88v56c0 35.7 22.5 72.4 61.9 100.7 31.5 22.7 69.8 37.1 110 41.7C203.3 338.5 240 360 240 360v72h-48c-35.3 0-64 20.7-64 56v12c0 6.6 5.4 12 12 12h296c6.6 0 12-5.4 12-12v-12c0-35.3-28.7-56-64-56h-48v-72s36.7-21.5 68.1-73.6c40.3-4.6 78.6-19 110-41.7 39.3-28.3 61.9-65 61.9-100.7V88c0-13.3-10.7-24-24-24zM99.3 192.8C74.9 175.2 64 155.6 64 144v-16h64.2c1 32.6 5.8 61.2 12.8 86.2-15.1-5.2-29.2-12.4-41.7-21.4zM512 144c0 16.1-17.7 36.1-35.3 48.8-12.5 9-26.7 16.2-41.8 21.4 7-25 11.8-53.6 12.8-86.2H512v16z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                        </span>
                      </h2>
                      <div
                        className="game-results_container"
                      >
                        <Row
                          noGutters={false}
                        >
                          <div
                            className="row"
                          >
                            <Col>
                              <div
                                className="col"
                              >
                                <div
                                  className="center"
                                >
                                  <FontAwesomeIcon
                                    border={false}
                                    className="centered-icon"
                                    fixedWidth={false}
                                    flip={null}
                                    icon={
                                      Object {
                                        "icon": Array [
                                          512,
                                          512,
                                          Array [],
                                          "f110",
                                          "M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z",
                                        ],
                                        "iconName": "spinner",
                                        "prefix": "fas",
                                      }
                                    }
                                    inverse={false}
                                    listItem={false}
                                    mask={null}
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size={null}
                                    spin={false}
                                    swapOpacity={false}
                                    symbol={false}
                                    title=""
                                    transform={null}
                                  >
                                    <svg
                                      aria-hidden="true"
                                      className="svg-inline--fa fa-spinner fa-w-16 centered-icon"
                                      data-icon="spinner"
                                      data-prefix="fas"
                                      focusable="false"
                                      role="img"
                                      style={Object {}}
                                      viewBox="0 0 512 512"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                                        fill="currentColor"
                                        style={Object {}}
                                      />
                                    </svg>
                                  </FontAwesomeIcon>
                                </div>
                              </div>
                            </Col>
                          </div>
                        </Row>
                      </div>
                    </div>
                  </div>
                </div>
              </GameResult>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Container>
</TicTacToe>
`;
